### PR TITLE
End of Python 2.7 support

### PR DESCRIPTION
**Pull request by dkfellows**
Also clean up a few other bits of bad practice.

### CODE REVIEW

File: examples/external_devices_examples/live_examples/spike_io.py [hunk 1]
```diff
@@ -30,8 +29,8 @@
 
 
 # Create an initialisation method
-def init_pop(label, n_neurons, run_time_ms, machine_timestep_ms):
-    print("{} has {} neurons".format(label, n_neurons))
+def init_pop(label, _n_neurons, run_time_ms, machine_timestep_ms):
```
Comment (alan-stokes) [inline examples/external_devices_examples/live_examples/spike_io.py]: why the _?

File: examples/partitioner_examples/splitter_usage.py [hunk 2]
```diff
@@ -15,13 +15,11 @@
...
-
-from pacman.model.partitioner_splitters import SplitterOneToOneLegacy
+from pacman.model.partitioner_splitters import (
+    SplitterOneToOneLegacy as OneToOneSplitter)
 from spynnaker.pyNN.extra_algorithms.splitter_components import (
-    SplitterAbstractPopulationVertexSlice)
-from spynnaker.pyNN.extra_algorithms.splitter_components.\
-    spynnaker_splitter_slice_legacy import (
-        SpynnakerSplitterSliceLegacy)
+    SplitterAbstractPopulationVertexSlice as APVSplitter,
```
Comment (alan-stokes) [inline examples/partitioner_examples/splitter_usage.py]: why rename? its not like you've saved any lines doing so

**Review by alan-stokes — APPROVED** [review]

File: examples/external_devices_examples/live_examples/spike_io.py (hunk 1 above)
Comment (Christian-B) [reply examples/external_devices_examples/live_examples/spike_io.py]: reverted

File: examples/partitioner_examples/splitter_usage.py (hunk 2 above)
Comment (Christian-B) [reply examples/partitioner_examples/splitter_usage.py]: reverted

**Pull request merged**